backend/src/services/group/group.service.ts

```diff
@@ -83,17 +83,29 @@ class Group {
     const permissions = await this.#permissionService.getByIds(permissionIds);
 
     if (permissions.items.length !== permissionIds.length) {
+      const permissionsNotExist = permissionIds.filter((id) => {
+        return !permissions.items.find((p) => p.id === id);
+      });
+      const message = `Permission with id ${permissionsNotExist.join(
+        ', ',
+      )} doesn't exist.` as ExceptionMessage;
       throw new GroupsError({
-        message: ExceptionMessage.INVALID_GROUP_PERMISSIONS,
+        message,
       });
     }
 
     if (userIds) {
       const users = await this.#userService.getByIds(userIds);
 
       if (users.length !== userIds.length) {
+        const usersNotExist = userIds.filter((id) => {
+          return !users.find((user) => user.id === id);
+        });
+        const message = `User with id ${usersNotExist.join(
```

Comment:
Not sure about the importance of the change I want to offer, but I think, that you should modify your error message in word "User" (`User${usersNotExist.length > 1 ?? 's'}`), because if your usersNotExist containt more than one element it will be grammatic error

Reply:
Same in the permissionsNotExist check

Reply:
Sorry, there should be User${usersNotExist.length > 1 && 's'}

Reply:
I was thinking about it but then would also need to do same with id/ids and don't/doesn't and the string would get noisy with checks

Reply:
You could create `getItemsNotExistsMessage` (or smth like that) helper to get rid of mass checks in service
```
const getSingularOrPluralWordFormatter = (isSingleElement: boolean) => (word: string): boolean => {
  return `${word}${isSingleElement ? 's' : ''}`;
};

const getItemsNotExistsMessage = <T> (items: T[], itemsName: string): string => {
  const getSingularOrPluralWord = getSingularOrPluralWordFormatter(items.length === 1);
  return `${getSingularOrPluralWord(itemsName)} with ${getSingularOrPluralWord('id')} ${items.join(', ')} not ${getSingularOrPluralWord('exist')}`;
};
```

Reply:
agreed

Reply:
agreed